lib/features/steps/feature_flag_steps.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+# @!group Feature flag steps
+
+# Verifies that there are no feature flags present in a given event
+#
+# @step_input event_id [Integer] The id of the event in the payloads array
+Then('event {int} has no feature flags') do |event_id|
+  event = Maze::Helper.read_key_path(Maze::Server.errors.current[:body], "events.#{event_id}")
+  assert_false(has_feature_flags?(event), "Feature flags expected to be absent or empty, was #{event['featureFlags']}")
+end
+
+# Verifies that the are no feature flags present
+Then('the event has no feature flags') do
+  steps %(
+    Then event 0 has no feature flags
+  )
+end
+
+# Verifies a feature flag with a specific variant is uniquely present in a givent even
+#
+# @step_input event_id [Integer] The id of the event in the payloads array
+# @step_input flag_name [String] The featureFlag value expected
+# @step_input variant [String] The variant value expected
+Then('event {int} contains the feature flag {string} with variant {string}') do |event_id, flag_name, variant|
+  event = Maze::Helper.read_key_path(Maze::Server.errors.current[:body], "events.#{event_id}")
+  assert(has_feature_flags?(event), "Expected feature flags were not present in event #{event_id}: #{event}")
+  feature_flags = event['featureFlags']
+  assert(
+    feature_flags.one? { |flag| flag['featureFlag'] == flag_name && flag['variant'] == variant },
+    "Feature flag: #{flag_name} with variant: #{variant} not found. Present flags: #{feature_flags}"
+  )
+end
+
+# Verifies a feature flag with a specific variant is uniquely present
+#
+# @step_input flag_name [String] The featureFlag value expected
+# @step_input variant [String] The variant value expected
+Then('the event contains the feature flag {string} with variant {string}') do |flag_name, variant|
+  steps %(
+    Then event 0 contains the feature flag #{flag_name} with variant #{variant}
+  )
+end
+
+# Verifies a feature flag with no variant (either null or missing) is uniquely present in a given event
+#
+# @step_input event_id [Integer] The id of the event in the payloads array
+# @step_input flag_name [String] The featureFlag value expected
+Then('event {int} contains the feature flag {string} with no variant') do |event_id, flag_name|
+  event = Maze::Helper.read_key_path(Maze::Server.errors.current[:body], "events.#{event_id}")
+  assert(has_feature_flags?(event), "Expected feature flags were not present in event #{event_id}: #{event}")
+  feature_flags = event['featureFlags']
+  assert(
+    feature_flags.one? { |flag| flag['featureFlag'] == flag_name && !flag.has_key?('variant') },
```

Comment:
This could allow duplicates if there are multiple flags with the same `featureFlag` but only one has no `variant`